backend/DatabaseScripts/SqlScripts/0039_CreateView_graphql_contracts_view.sql

```diff
@@ -0,0 +1,108 @@
+﻿ALTER TABLE graphql_transaction_events
+	ALTER COLUMN event SET DATA type jsonb;
+CREATE INDEX event_tag on graphql_transaction_events (((event->>'tag')::INTEGER));
+CREATE INDEX event_contract_init_contract_address ON graphql_transaction_events (((event->'data'->>'ContractAddress'))) WHERE (event->>'tag')::INTEGER = 16;
+CREATE INDEX event_contract_upgrade_contract_address ON graphql_transaction_events (((event->'data'->>'ContractAddress'))) WHERE (event->>'tag')::INTEGER = 36;
+CREATE INDEX event_contract_update_contract_address ON graphql_transaction_events (((event->'data'->>'ContractAddress'))) WHERE (event->>'tag')::INTEGER = 18;
+CREATE INDEX event_contract_transfer_from_contract_address ON graphql_transaction_events (((event->'data'->'From'->>'data'))) WHERE (event->>'tag')::Integer = 1 AND (event->'data'->'From'->>'tag')::Integer = 2;
+CREATE OR REPLACE VIEW graphql_contracts_view 
+AS
+	WITH contract_init_events AS (
+	SELECT
+		transaction_id,
+		index,
+		(event->'data'->>'ContractAddress') as contract_address,
+		(event->'data'->>'ModuleRef') as module_ref,
+		(event->'data'->>'Amount')::BIGINT as amount_in
+	FROM graphql_transaction_events 
+	WHERE (event->>'tag')::Integer = 16
+	),
+	contract_upgrade_events AS (
+		SELECT
+			transaction_id,
+			index,
+			(event->'data'->>'ContractAddress') as contract_address,
+			(event->'data'->>'From') as from_module_ref,
+			(event->'data'->>'To') as to_module_ref
+		FROM graphql_transaction_events
+		WHERE (event->>'tag')::Integer = 36
+	),
+	contract_update_events AS (
+			SELECT
+			transaction_id,
+			index,
+			(event->'data'->>'ContractAddress') as contract_address,
+			(event->'data'->>'Amount')::BIGINT as amount_in
+		FROM graphql_transaction_events
+		WHERE (event->>'tag')::Integer = 18
+	),
+	contract_transfer_events AS (
+		SELECT
+			transaction_id,
+			index,
+			(event->'data'->'From'->>'data') as contract_address,
+			(event->'data'->>'Amount')::BIGINT as amount_out
+		FROM graphql_transaction_events
+		WHERE (event->>'tag')::Integer = 1 AND (event->'data'->'From'->>'tag')::Integer = 2
+	),
+	contract_module AS (
+		SELECT contract_address, module_ref, first_transaction_id, last_transaction_id, transactions_count FROM (
+			SELECT 
+				FIRST_VALUE(transaction_id) OVER W1 AS first_transaction_id, 
+				LAST_VALUE(transaction_id) OVER W1 AS last_transaction_id, 
+				contract_address, 
+				module_ref,
+				COUNT(transaction_id) over W1 AS transactions_count,
+				rank() OVER W1 as rn
+			FROM (
+				SELECT transaction_id, index, contract_address, module_ref FROM contract_init_events
+				UNION ALL 
+				SELECT transaction_id, index, contract_address, to_module_ref as module_ref FROM contract_upgrade_events
+			) e1
+			WINDOW w1 AS (PARTITION BY contract_address ORDER BY transaction_id DESC)
+		) e2
+		WHERE rn = 1
```

Comment:
I would look into this since one create a whole table and one uses one row - seem like other grouping should be done.